code/modules/mob/living/simple_animal/hostile/illusion.dm

```diff
@@ -38,13 +41,70 @@
 	transform = initial(transform)
 	pixel_y = initial(pixel_y)
 	pixel_x = initial(pixel_x)
+	fake_huds()
+	add_to_all_human_data_huds() //For testing the spawn no shooting
 
 /mob/living/simple_animal/hostile/illusion/examine(mob/user)
 	if(parent_mob)
 		. = parent_mob.examine(user)
 	else
 		. = ..()
 
+/mob/living/simple_animal/hostile/illusion/proc/fake_huds()
+	if(!parent_mob || !ishuman(parent_mob))
+		return
+	var/mob/living/carbon/human/H = parent_mob
+	var/image/holder = hud_list[ID_HUD]
+	holder.icon_state = "hudunknown"
+	if(H.wear_id)
+		holder.icon_state = "hud[ckey(H.wear_id.get_job_name())]"
+	var/image/holder2
+	for(var/i in list(IMPTRACK_HUD, IMPMINDSHIELD_HUD, IMPCHEM_HUD))
+		holder2 = hud_list[i]
+		holder2.icon_state = null
+	for(var/obj/item/implant/I in H)
+		if(I.implanted)
+			if(istype(I,/obj/item/implant/tracking))
+				holder2 = hud_list[IMPTRACK_HUD]
+				holder2.icon_state = "hud_imp_tracking"
+			else if(istype(I,/obj/item/implant/mindshield))
+				holder2 = hud_list[IMPMINDSHIELD_HUD]
+				holder2.icon_state = "hud_imp_loyal"
+			else if(istype(I,/obj/item/implant/chem))
+				holder2 = hud_list[IMPCHEM_HUD]
+				holder2.icon_state = "hud_imp_chem"
+	var/image/holder3 = hud_list[WANTED_HUD]
+	var/perpname = H.get_visible_name(TRUE) //gets the name of the perp, works if they have an id or if their face is uncovered
+	if(!SSticker) return //wait till the game starts or the monkeys runtime....
```

Comment:
is this line copy pasted?
```dm
if(!SSticker)
   return
```
doesn't work anymore because SSticker will never be null
instead you need to perform this check
```suggestion
	if(SSticker.current_state != GAME_STATE_PLAYING)
		return //wait till the game starts or the monkeys runtime....
```
also, it'd probably make sense to move this check up to the guard clause on L54 so you don't need to do all that code above only to do nothing with it

Reply:
Done!